packages/pagination/__tests__/pagination.spec.ts

```diff
@@ -1,407 +1,377 @@
-    expect(wrapper.find('ul.el-pager').exists()).toBe(true)
-    expect(wrapper.find('button.btn-next').exists()).toBe(true)
-    expect(wrapper.find('.el-pagination__rightwrapper').exists()).toBe(true)
-    expect(wrapper.find('.el-pagination__rightwrapper .el-pagination__jump').exists()).toBe(true)
-
-    layout.value = 'prev, pager, next, ->, jumper, sizes'
-    await nextTick()
-    expect(wrapper.find('button.btn-prev').exists()).toBe(true)
-    expect(wrapper.find('ul.el-pager').exists()).toBe(true)
-    expect(wrapper.find('button.btn-next').exists()).toBe(true)
-    expect(wrapper.find('.el-pagination__rightwrapper').exists()).toBe(true)
-    expect(wrapper.find('.el-pagination__rightwrapper .el-pagination__jump').exists()).toBe(true)
-    expect(wrapper.find('.el-pagination__rightwrapper .el-pagination__sizes').exists()).toBe(true)
-
-    layout.value = 'prev, pager, next, ->, jumper, sizes, total'
-    await nextTick()
-    expect(wrapper.find('button.btn-prev').exists()).toBe(true)
-    expect(wrapper.find('ul.el-pager').exists()).toBe(true)
-    expect(wrapper.find('button.btn-next').exists()).toBe(true)
-    expect(wrapper.find('.el-pagination__rightwrapper').exists()).toBe(true)
-    expect(wrapper.find('.el-pagination__rightwrapper .el-pagination__jump').exists()).toBe(true)
-    expect(wrapper.find('.el-pagination__rightwrapper .el-pagination__sizes').exists()).toBe(true)
-    expect(wrapper.find('.el-pagination__rightwrapper .el-pagination__total').exists()).toBe(true)
-  })
-
-  test('slot', () => {
-    const TestComponent = {
-      template: `
-        <el-pagination
-          layout="slot, prev, pager, next"
-          :page-size="25"
-          :total="100">
-          <span class="slot-test">slot test</span>
-        </el-pagination>
-      `,
-      components: {
-        'el-pagination': Pagination,
-      },
-    }
-    const wrapper = mount(TestComponent)
-    expect(wrapper.find('.slot-test').exists()).toBe(true)
-  })
-
-  test('small', () => {
-    const wrapper = mount(Pagination, {
-      props: {
-        small: true,
-      },
-    })
-    expect(wrapper.vm.$el.classList.contains('el-pagination--small')).toBe(true)
-  })
-
-  test('pageSize', () => {
-    const wrapper = mount(Pagination, {
-      props: {
-        pageSize: 25,
-        total: 100,
-      },
     })
-    expect(wrapper.findAll('li.number').length).toBe(4)
-  })
-
-  test('pageSize: NaN', () => {
-    const wrapper = mount(Pagination, {
-      props: {
-        pageSize: NaN,
-        total: 100,
-      },
+    test('layout empty', async () => {
+      await nextTick()
+      expect(wrapper.find('.el-pagination').exists()).toBe(false)
     })
-    expect(wrapper.findAll('li.number').length).toBe(7)
-  })
-
-  test('pageCount', () => {
-    const wrapper = mount(Pagination, {
-      props: {
-        pageSize: 25,
-        pagerCount: 5,
-        pageCount: 50,
-      },
+    const layoutSelectorCombines = [
```

Comment:
Maybe `combinator` is a more precise word here IMHO.

Reply:
You inspiring me with a better name `layoutSelectorPairs`